operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-querier-schemas-configmap.yaml

```diff
@@ -0,0 +1,22 @@
+{{- if .Values.graphite.enabled -}}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "mimir.fullname" . }}-graphite-schemas
+  labels:
+    app: {{ template "mimir.name" . }}
+    chart: {{ template "mimir.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+  namespace: {{ .Release.Namespace | quote }}
+data:
+{{- range  $storageSchemas := toStrings .Values.graphite_querier.schemasConfiguration.storageSchemas }}
```

Comment:
I'm getting an error when I attempt this: 
```
Error: UPGRADE FAILED: YAML parse error on mimir-distributed/templates/graphite-proxy/graphite-querier-schemas-configmap.yaml: error converting YAML to JSON: yaml: line 16: could not find expected ':'
```
What I wrote: 
```
data:
  storage-schemas.conf: {{ .Values.graphite_querier.schemasConfiguration.storageSchemas }}
  storage-aggregations.conf: {{ .Values.graphite_querier.schemasConfiguration.storageAggregations }}
```

This is what its producing: 
```
# Source: mimir-distributed/templates/graphite-proxy/graphite-queri
er-schemas-configmap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: graphite-demo-mimir-graphite-schemas
  labels:
    helm.sh/chart: mimir-distributed-3.1.0-weekly.197
    app.kubernetes.io/name: mimir
    app.kubernetes.io/instance: graphite-demo
    app.kubernetes.io/component: graphite-schemas
    app.kubernetes.io/version: "r197"
    app.kubernetes.io/managed-by: Helm
  namespace: "default"
data:
  storage-schemas.conf: [default]
pattern = .*
intervals = 0:1s
retentions = 10s:8d,10min:1y
  storage-aggregations.conf: [default]
aggregationMethod = avg
pattern = .*
xFilesFactor = 0.1
---
```

I think that because its a multiline string that is unquoted it fails to parse. This is why i was using range + toStrings.

Reply:
try using 
```
data:
  storage-schemas.conf: {{ .Values.graphite_querier.schemasConfiguration.storageSchemas | indent 2 }}
  storage-aggregations.conf: {{ .Values.graphite_querier.schemasConfiguration.storageAggregations | indent 2 }}
```

helm does not understand that the surrounding file needs to be valid yaml, so it just literally inlines a block of text. you need to tell it to indent it by 2 so it doesnt break the yaml structure

Reply:
Thanks, make it work with this:
```
  storage-schemas.conf: |
{{ .Values.graphite_querier.schemasConfiguration.storageSchemas | indent 4 }}
  storage-aggregations.conf: |
{{ .Values.graphite_querier.schemasConfiguration.storageAggregations | indent 4 }}
```